dev/Notifications/BadgeNotifications/BadgeNotificationManager.cpp

```diff
@@ -0,0 +1,157 @@
+        SetBadge(glyphValueString, nullptr);
+        return;
+    }
+
+    void BadgeNotificationManager::SetBadgeAsGlyph(winrt::BadgeNotificationGlyph glyphValue, winrt::DateTime expiration)
+    {
+        THROW_HR_IF_MSG(E_ILLEGAL_METHOD_CALL, !AppModel::Identity::IsPackagedProcess(), "Not applicable for unpackaged applications");
+
+        PCWSTR glyphValueString;
+        GetBadgeNotificationGlyphToString(glyphValue, &glyphValueString);
+        SetBadge(glyphValueString, &expiration);
+        return;
+    }
+
+    void BadgeNotificationManager::ClearBadge()
+    {
+        THROW_HR_IF_MSG(E_ILLEGAL_METHOD_CALL, !AppModel::Identity::IsPackagedProcess(), "Not applicable for unpackaged applications");
+
+        auto logTelemetry{ BadgeNotificationTelemetry::ClearBadge::Start(
+            g_telemetryHelper,
+            m_appId) };
+
+        BaseNotifications::BaseNotificationManager::RemoveAllNotification(ToastABI::NotificationType::NotificationType_Badge);
+
+        logTelemetry.Stop();
+
+        return;
+    }
+
+    void BadgeNotificationManager::GetBadgeNotificationGlyphToString(_In_ winrt::BadgeNotificationGlyph glyphValue, _Out_ PCWSTR* glyphString)
+    {
+        static const std::unordered_map<winrt::BadgeNotificationGlyph, PCWSTR> enumMapping = {
+            {winrt::BadgeNotificationGlyph::None, L"none"},
+            {winrt::BadgeNotificationGlyph::Activity, L"activity"},
+            {winrt::BadgeNotificationGlyph::Alarm, L"alarm"},
+            {winrt::BadgeNotificationGlyph::Alert, L"alert"},
+            {winrt::BadgeNotificationGlyph::Attention, L"attention"},
+            {winrt::BadgeNotificationGlyph::Available, L"available"},
+            {winrt::BadgeNotificationGlyph::Away, L"away"},
+            {winrt::BadgeNotificationGlyph::Busy, L"busy"},
+            {winrt::BadgeNotificationGlyph::Error, L"error"},
+            {winrt::BadgeNotificationGlyph::NewMessage, L"newMessage"},
+            {winrt::BadgeNotificationGlyph::Paused, L"paused"},
+            {winrt::BadgeNotificationGlyph::Playing, L"playing"},
+            {winrt::BadgeNotificationGlyph::Unavailable, L"unavailable"}
+        };
+
+        *glyphString = enumMapping.at(glyphValue);
+        return;
+    }
+
+    void BadgeNotificationManager::SetBadge(_In_ const std::wstring& value, _In_opt_ const winrt::Windows::Foundation::DateTime* expiration)
+    {
+        auto xmlResult = wil::str_printf<std::wstring>(L"<badge value='%ls'/>", value.c_str());
+        ::Microsoft::Windows::BadgeNotifications::BadgeNotification badgeNotifications(xmlResult.c_str());
+
+        auto logTelemetry{ BadgeNotificationTelemetry::SetBadge::Start(
+            g_telemetryHelper,
+            m_appId,
+            badgeNotifications.Payload().c_str()) };
+
+        if (expiration != nullptr)
+        {
+            badgeNotifications.Expiration(*expiration);
+        }
+
+        BaseNotifications::BaseNotificationManager::Show(badgeNotifications);
```

Comment:
Feels like this interplay of the manager, base, and notification types needs to be simplified since the BadgeNotification types are no longer returned to anybody. Maybe this "show" method should also just take `xmlResult.c_str()` + `expiration ? *expiration : {}` ?

Reply:
removed baseNotificationManager but keeping BaseNotification because it simplifies the already existing Notification code. E.g - AppNotification now inherits BaseNotification and BadgeNotification also inherits BaseNotification. Otherwise duplicity of code would have happened. Also this makes both AppNotification and BadgeNotification extensible and easy to add new attributes